File: frontend/src/features/recipes/services/recipeService.js
import { apiRequest, toQueryString } from '../../../lib/apiClient'

export function listRecipes(params = { page: 1, pageSize: 20 }) {
  return apiRequest(`/recipes${toQueryString(params)}`)
}

export function getRecipe(id) {
  return apiRequest(`/recipes/${id}`)
}

export function createRecipe(data) {
  return apiRequest('/recipes', { method: 'POST', body: data })
}

export function updateRecipe(id, data) {
  return apiRequest(`/recipes/${id}`, { method: 'PUT', body: data })
}

export function deleteRecipe(id) {
  return apiRequest(`/recipes/${id}`, { method: 'DELETE' })
}

export function importRecipeFromLink(sourceUrl) {
  return apiRequest('/recipes/import/video', { method: 'POST', body: { sourceUrl } })
}

File: backend/src/main/java/com/recipemanager/recipes/application/RecipeService.java
package com.recipemanager.recipes.application;

import com.recipemanager.recipes.api.RecipeListResponse;
import com.recipemanager.recipes.api.RecipeRequest;
import com.recipemanager.recipes.api.RecipeResponse;
import java.util.UUID;

public interface RecipeService {

    RecipeListResponse listRecipes(UUID userId, int page, int pageSize);

    RecipeResponse getRecipe(UUID userId, UUID id);

    RecipeResponse createRecipe(UUID userId, RecipeRequest request);

    RecipeResponse updateRecipe(UUID userId, UUID id, RecipeRequest request);

    void deleteRecipe(UUID userId, UUID id);
}